File: Greedy_Algorithm/Python/백준_그리디알고리즘_동전0(11047).py
N, K = map(int,input().split())
money = []
count = 0
for _ in range(N):
    Input = int(input())
    money.append(Input)
money.sort(reverse=True)
for i in range(N):
    if(K>=money[i]):
        count += int(K/money[i])
        K %= money[i]
    else: continue
print(count)

File: Greedy_Algorithm/C++/백준_그리디알고리즘_동전0(11047).cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    int N, K;
    int money;
    int count = 0;
    vector <int> vec;
    cin >> N >> K;
    for (int i = 0; i < N; i++) {
        cin >> money;
        vec.push_back(money);
    }
    sort(vec.begin(), vec.end(), greater<>());
    for (int i = 0; i < N; i++) {
        if (K >= vec[i]) {
            count += K / vec[i];
            K %= vec[i];
        }
        else continue;
    }
    cout << count << endl;
    return 0;
}